src/uwtools/tests/config/test_core.py

```diff
@@ -326,8 +376,8 @@ def test_ini_config_bash(salad_base, tmp_path):
     """
     infile = fixture_path("simple.sh")
     outfile = tmp_path / "outfile.sh"
-    cfgobj = config.INIConfig(infile, space_around_delimiters=False)
-    expected = {**salad_base["salad"], "how_many": "12"}  # str "12" (not int 12) for INI
+    cfgobj = core.INIConfig(infile, space_around_delimiters=False)
+    expected: dict = {**salad_base["salad"], "how_many": "12"}  # str "12" (not int 12) for INI
```

Comment:
Just for my curiosity...why is the typing needed here? It seemed pretty obvious that this would be come a dict, and NOT a set.

Reply:
Good question! This was initially a puzzle for me, too (a puzzle you can't solve because you don't have all the info in this PR.) The issue isn't actually on this line, but a little further down:
```
expected["dressing"] = ["ranch", "italian"]
```
On the line you asked about, a typechecker can infer that the type of `expected` is `List[str, str]`, since `salad_base` is initially a `List[str, Union[str, int]]` (because `how_many` is an `int` in the original fixture value) that becomes a `List[str]` because `how_many` is turned into a `str` on this line, for compatibility with INI formatting. Then the addition of the `list` value `["ranch", "italian"]` violates the `List[str]` type, which would have to be `List[Union[str, list]]` (or even, if you want to be a total type nerd, `List[Union[str, List[str]]]`) to be correct.

So I typed `expected` as `dict`, which is a synonym for `Dict[Any, Any]`. But I'm going to update the type annotation to `Dict[str, Any]` in hopes that that will give a better hint to future readers about what's going on. Thank you for asking about this.

BTW, `mypy` didn't complain about the original, un-annotated code, but `pyright`, which I use in my editor did. VSCode users probably would have seen the same complaint since `pyright` is Microsoft's Python typechecker and presumably what their VSCode uses, and in case `mypy` starts complaining in a future release, I thought I'd be proactive.

Reply:
Thanks again for the explanation! An interesting outcome I would not have considered having nearly no first-hand experience with the types so far!